test/Mono.Linker.Tests.Cases.Expectations/Assertions/ProducedBy.cs

```diff
@@ -9,7 +9,10 @@ namespace Mono.Linker.Tests.Cases.Expectations.Assertions
 	public enum ProducedBy
 	{
 		Trimmer = 1,
-		Analyzer = 2,
+		RequiresAssemblyFileAnalyzer = 2,
```

Comment:
This seems like it's leaking implementation details. We shouldn't need to know which analyzer produces it, only if it's a diagnostic currently produced by any of the analyzers.

Reply:
But then that would be a problem whenever some test suite in Mono.Linker.Tests.Cases has `ExpectedWarning` with warning codes for different analyzers, no? e.g., say you have A.cs with:

```C#
[ExpectedWarning("ILXXXX", ProducedBy = ProducedBy.Analyzer)] // This is only produced by analyzer A
[ExpectedWarning("ILYYYY", ProducedBy = ProducedBy.Analyzer)] // This is only produced by analyzer B
void Foo () { ... }
```

Since analyzer tests are run separately, when running the tests in A.cs with analyzer A, we'd fail since it wouldn't produce ILYYYY; similar when running A.cs tests with B, in which case we'd be missing ILXXXX.

Reply:
We could also allow to pass multiple analyzers to `RunTest` in the analyzer test infra, if that makes more sense than having all analyzers in the enum.

Reply:
Yeah, it is leaking details a bit due to the impedance mismatch between the linker/analyzer test infrastructure - running with all the analyzers would be good to avoid this. But I like this approach as a way to ignore some expected warnings when testing with only one analyzer for now. Maybe it's worth a comment here to say why we have the individual enum values.